### PR TITLE
Voltage control/Smartmax

### PR DESCRIPTION
**Pull request by EyeCantCU**
This enables voltage control for both Onyx and Bacon. While I was at it, I also added adaptive LMK and Smartmax as well.

### CODE REVIEW

**Comment by xboxfanj** [comment]
Looks good.  I'll test in a bit.  One thing you may want to do in the future is autogenerate the defconfig after making changes.  It's just a good practice to get into for organization so that everything is always in sync.

To do this, follow the steps below.

cd to the kernel directory

export ARCH=arm

make baconx_defconfig

make savedefconfig

mv defconfig arch/arm/configs/baconx_defconfig

**Comment by EyeCantCU** [comment]
Okay. Thanks for the information. I'll do that in the future.

**Comment by EyeCantCU** [comment]
I regenerated the defconfig, and today I added Smartmax and adaptive LMK. I will wait in the future for pull requests to be merged before committing :)

File: arch/arm/boot/dts/msm8974-v2.dtsi [hunk 1]
```diff
@@ -123,7 +123,7 @@
 		<1880000 2068000>,
 		<3008000 3309000>,
 		<3760000 4136000>,
-		<4468000 2457000>;
+		<4468000 2457600>;
```
Comment (xboxfanj) [inline arch/arm/boot/dts/msm8974-v2.dtsi]: Neither of these files are used.  OnePlus separates out their dts files into arch/arm/boot/dts/project

In that directory, 14001_HW is for OPO, and 15055_HW is for OPX.

**Comment by xboxfanj** [comment]
Will review when I have a chance.  In the future, please use one pull request for one feature (one for SmartMAX, one for Adaptive LMK, one for Voltage Control).

**Comment by EyeCantCU** [comment]
Okay

File: arch/arm/boot/dts/msm8974-v2.dtsi (hunk 1 above)
Comment (EyeCantCU) [inline arch/arm/boot/dts/msm8974-v2.dtsi]: Okay. I will revert that

Comment (EyeCantCU) [inline arch/arm/boot/dts/msm8974-v2.dtsi]: As far as DTS files, isn't it vice versa? 14001 only goes up to 2.265ghz where as 15055 goes to 2.4ghz. I'm just curious in this regard. (msm8974pro.dtsi)

**Comment by EyeCantCU** [comment]
Went ahead and removed everything dealing with adaptive LMK. In the future, my pull requests will be separate

**Comment by xboxfanj** [comment]
Thank you :)

**Comment by EyeCantCU** [comment]
get_cpu_idle_time has to be declared in cpufreq.c @scafroglia93 and Interactive must have its arguments changed so that it does not conflict. I overlooked this during the pull request and well... it was all reverted anyway... But what do I know anyway? I'm just here to learn.

This should do it:
https://github.com/sultanxda/android_kernel_oneplus_onyx/commit/cf01e5436cb56fc0e7696b31c1aceaa0c62b91fc

My sources have this fixed already and will stay that way along with CPU input boost as it does correlate to better battery life on my end... Sorry for all the trouble I've caused everyone. I think I am done here.

**Pull request merged**